### PR TITLE
Add (limited) support for install via composer-merge-plugin

### PR DESCRIPTION
**Pull request by filbertkm**
Things like generating WikibaseClient.settings.php don't work yet with the plugin, but for a general install that should be okayish.  added a detailed @todo to address that.

Bug: [T95673](https://phabricator.wikimedia.org/T95673)

### CODE REVIEW

**Comment by filbertkm** [comment]
I have tested this with composer-merge-plugin. I want to double check and manually test again with the build process and vagrant, before we merge this.

File: Wikidata.php [hunk 1]
```diff
@@ -18,18 +18,38 @@
...
+	// composer-merge-plugin
+	if ( !empty( $wmgUseWikibaseRepo ) ) {
+		include_once __DIR__ . '/WikibaseRepo.settings.php';
+	}
+
+	if ( !empty( $wmgUseWikibaseClient ) ) {
+		include_once __DIR__ . '/WikibaseClient.settings.php';
+	}
+
+	$wgWikidataBaseDir = __DIR__;
```
Comment (thiemowmde) [inline Wikidata.php]: I suggest to move this line to the top of the `if`, just as in the other file, for readability.

**Comment by thiemowmde** [comment]
+1, code looks good.

**Comment by JanZerebecki** [comment]
Tested in vagrant and manually triggered a build.

**Pull request merged**